modules/virt-creating-a-service-from-a-virtual-machine.adoc

```diff
@@ -0,0 +1,232 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/vm_networking/virt-using-the-default-pod-network-with-virt.adoc
+
+[id="virt-creating-a-service-from-a-virtual-machine_{context}"]
+
+= Creating a service from a virtual machine
+
+Create a service from a running virtual machine by first creating a Service object to expose the virtual machine.
+
+The `ClusterIP` service type exposes the virtual machine internally, within the cluster. The `NodePort` or `LoadBalancer` service types expose the virtual machine externally, outside of the cluster.
+
+This procedure presents an example of how to create, connect to, and expose a Service object of `type: ClusterIP` as a virtual machine-backed service.
+
+[NOTE]
+====
+`ClusterIP` is the default service `type`, if the service `type` is not specified.
+====
+
+.Procedure
+
+. Edit the virtual machine YAML as follows:
++
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  name: vm-ephemeral
+  namespace: example-namespace
+spec:
+  running: false
+  template:
+    metadata:
+      labels:
+        special: key <1>
+    spec:
+      domain:
+        devices:
+          disks:
+            - name: containerdisk
+              disk:
+                bus: virtio
+            - name: cloudinitdisk
+              disk:
+                bus: virtio
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 1024M
+      networks:
+        - name: default
+          pod: {}
+      volumes:
+        - name: containerdisk
+          containerDisk:
+            image: kubevirt/fedora-cloud-container-disk-demo
+        - name: cloudinitdisk
+          cloudInitNoCloud:
+            userData: |
+              #!/bin/bash
+              echo "fedora" | passwd fedora --stdin
+----
+<1> Add the label `special: key` in the `spec.template.metadata.labels` section.
++
+
+[NOTE]
+====
+Labels on a virtual machine are passed through to the pod. The labels on
+the `VirtualMachine`, for example `special: key`, must match the labels in
+the Service  YAML `selector` attribute, which you create later
+in this procedure.
+====
+
+. Save the virtual machine YAML to apply your changes.
+
+. Edit the service YAML to configure the settings necessary to create and expose the Service object.
```

Comment:
s/object./object:/

Reply:
Fixed. Still kind of confused about when to use colons as I have been told different things by other reviewers. What I have done in the past is only use them if the sentence says something like "Edit the service YAML as follows" (a direct reference). A lot of editors I've worked with in the pas are just allergic to colons in general, I think. I have no specific preference, just trying to get a handle on what is expected... :-)

Reply:
Gotcha. Yeah I noticed the difference with the "as follows:" earlier. I've typically just always had lead-in colons for any command/action we're telling them to do directly following. Worth bringing up to get clarity between reviewers/the team, though, for sure.